packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,192 @@
+import SelectorEngine from './dom/SelectorEngine';
+import BaseComponent from './BaseComponent';
+import { enableToggleAutoloader } from './utils';
+
+const NAME = 'fileUploader';
+const IS_DRAGGABLE_CLASS_NAME = 'has-drag-and-drop';
+const IS_DRAGGING_CLASS_NAME = 'is-dragging';
+const FILE_SIZE_LIMIT = 10000000; // = 10mb
+
+class FileUploader extends BaseComponent {
+  element: HTMLElement;
+  wrapper: HTMLElement;
+  inputElement: HTMLInputElement;
+  listElement: HTMLElement | null;
+  dropZone: HTMLElement | null;
+  isDraggable: boolean;
+  isMultiple: boolean;
+  isDragging: boolean;
+  fileQueue: File[];
+
+  constructor(element: HTMLElement) {
+    super(element);
+
+    this.element = element;
+    this.wrapper = SelectorEngine.findOne('.FileUploaderInput', this.element) as HTMLInputElement;
+    this.inputElement = SelectorEngine.findOne('.FileUploaderInput__input', this.element) as HTMLInputElement;
+    this.listElement = SelectorEngine.findOne('.FileUploaderList', this.element);
+    this.dropZone = SelectorEngine.findOne('.FileUploaderInput__dropZone', this.element);
+    this.isDraggable = 'draggable' in document.createElement('span');
+    this.isMultiple = this.inputElement.multiple;
+    this.isDragging = false;
+    this.fileQueue = [];
+
+    this.init();
+    this.getFileQueue.bind(this);
+    this.clearFileQueue.bind(this);
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  public getFileQueue() {
+    return this.fileQueue;
+  }
+
+  public clearFileQueue() {
+    this.fileQueue = [];
+    this.listElement!.textContent = '';
+  }
+
+  appendToList(file: File) {
+    if (!this.listElement) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Unfortunately, there is no list element created for inserting files to upload ⚠️');
```

Comment:
No, this is an element. The message is displayed only if the sheet element does not exist. The queue list is something else.

Reply:
So, the developer is responsible for inserting this empty list into the DOM? Am I right? Can we do this for him/her? E.g. insert the list when the first item is "uploaded"...?

Reply:
I originally planned it that way, but because part of the sheet is also a hidden title with an identifier. So we would have to set the embedding properties in a complex way.